src/main/java/org/elasticsearch/common/io/PathUtils.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.io;
+
+import org.elasticsearch.common.SuppressForbidden;
+
+import java.net.URI;
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/** 
+ * Utilities for creating a Path from names,
+ * or accessing the default FileSystem.
+ * <p>
+ * This class allows the default filesystem to
+ * be changed during tests.
+ */
+@SuppressForbidden(reason = "accesses the default filesystem by design")
+public final class PathUtils {
+    /** no instantiation */
+    private PathUtils() {}
+    
+    /** the actual JDK default */
+    static final FileSystem ACTUAL_DEFAULT = FileSystems.getDefault();
+    
+    /** can be changed by tests */
+    static FileSystem DEFAULT = ACTUAL_DEFAULT;
```

Comment:
I'm confused what forbidden can do here. its package-private (test class sets with reflection). And its ok to ask for the default filesystem, that is why we have a getter.